Web app start fix when config file is missing #1413

ozankiraz opened the pull request:
Created New PR since I messed up previous one

tulinkry commented on web/index.jsp:
```diff
@@ -29,6 +29,14 @@ include file="projects.jspf"
+    throw new java.io.FileNotFoundException("Configuration File Not Found");
+    }
+    File sourceRootPathFile = RuntimeEnvironment.getInstance().getSourceRootFile();
+    if(sourceRootPathFile.exists() && !sourceRootPathFile.canRead()){
```
This is not correct
```java
if(sourceRootPathFile.exists() && !sourceRootPathFile.canRead()){
```

Perhaps should have been:
```java
if(!sourceRootPathFile.exists() || !sourceRootPathFile.canRead()){
```

ozankiraz replied:
@tulinkry I intended to do this actually, There are two exceptions, first one is FileNotFoundException, thrown when source path is null or empty, covers the scenario when source root path is not set,
Second exception is IOException, covers the scenario when file is there but cannot read

tulinkry replied:
But when the file does not actually exist then there is no exception which is not correct, is it?

ozankiraz commented on src/org/opensolaris/opengrok/web/PageConfig.java:
```diff
@@ -1402,4 +1402,20 @@ public String getPathTitle() {
         return Util.htmlize(title);
     }
+    
+    public void checkSourceRootExistence() throws IOException {        
```
Moved logic here

ozankiraz commented on src/org/opensolaris/opengrok/web/PageConfig.java:
```diff
@@ -1402,4 +1402,20 @@ public String getPathTitle() {
+            throw new java.io.IOException("Can not read configuration file");
+        }
+        
+        if(!sourceRootPathFile.exists()){
```
Added this check as required

ozankiraz commented on web/index.jsp:
```diff
@@ -29,14 +29,7 @@ include file="projects.jspf"
-    if(sourceRootPathFile.exists() && !sourceRootPathFile.canRead()){
-    throw new java.io.IOException("Can not read configuration file");
-    }
+    cfg.checkSourceRootExistence();
```
Shorter and more readable

ozankiraz commented:
@tulinkry Yes, you're right, I was confused about SRC_ROOT in web.xml and sourceRoot in configuration.xml
Moved logic to PageConfig class and updated jsp file. Thanks for pointing it out.

tulinkry commented:
Ok to me now

vladak commented:
@ozankiraz any news w.r.t. your OCA ?

ozankiraz commented:
Hi @vladak 
I got an email about OCA is processed and I can see my name in OCA list, but in email they didn't sent me OCA number.

vladak commented:
Okay, I see your name there, merging.
